core/lib/scmi-server/sub.mk

```diff
@@ -0,0 +1,174 @@
+		-Wno-missing-prototypes \
+		-Wno-missing-declarations \
+		-Wno-suggest-attribute=format \
+		-Wno-declaration-after-statement
+
+# The below directives will be removed once SCP-firmware pull requests
+# 728, 729 and 730 are merged.
+cflags-lib-y += -Wno-undef \
+		-Wno-maybe-uninitialized \
+		-Wno-unused-but-set-variable \
+		-Wno-suggest-attribute=format \
+
+cflags-lib-$(_CFG_SCPFW_NOTIFICATION) += -DBUILD_HAS_NOTIFICATION \
+					 -DBUILD_HAS_SCMI_NOTIFICATION
+
+cflags-lib-$(_CFG_SCPFW_FAST_CHANNELS) += -DBUILD_HAS_FAST_CHANNELS \
+					  -DBUILD_HAS_SCMI_FAST_CHANNELS
+
+incdirs_ext-y += $(scpfw-path)/arch/none/optee/include
+srcs-y += $(scpfw-path)/arch/none/optee/src/arch_interrupt.c
+srcs-y += $(scpfw-path)/arch/none/optee/src/arch_main.c
+
+incdirs_ext-y += $(scpfw-path)/framework/include
+srcs-y += $(scpfw-path)/framework/src/fwk_arch.c
+srcs-y += $(scpfw-path)/framework/src/fwk_dlist.c
+srcs-y += $(scpfw-path)/framework/src/fwk_id.c
+srcs-y += $(scpfw-path)/framework/src/fwk_interrupt.c
+srcs-y += $(scpfw-path)/framework/src/fwk_io.c
+srcs-y += $(scpfw-path)/framework/src/fwk_log.c
+srcs-y += $(scpfw-path)/framework/src/fwk_mm.c
+srcs-y += $(scpfw-path)/framework/src/fwk_module.c
+srcs-y += $(scpfw-path)/framework/src/fwk_ring.c
+srcs-y += $(scpfw-path)/framework/src/fwk_slist.c
+srcs-y += $(scpfw-path)/framework/src/fwk_status.c
+srcs-y += $(scpfw-path)/framework/src/fwk_string.c
+srcs-y += $(scpfw-path)/framework/src/fwk_delayed_resp.c
+srcs-y += $(scpfw-path)/framework/src/fwk_time.c
+srcs-y += $(scpfw-path)/framework/src/fwk_core.c
+srcs-y += $(scpfw-path)/framework/src/assert.c
+srcs-y += $(scpfw-path)/framework/src/stdlib.c
+srcs-$(_CFG_SCPFW_NOTIFICATION) += $(scpfw-path)/fwk_notification.c
+
+# Helper macros for listing SCP-firmware modules source files (in srcs-y)
+# and header include path (in incdirs_ext-y). Each module provide a C source
+# file named mod_<module-name>.c and possibly an include directory. Build
+# directive BUILD_HAS_MOD_<NAME> must be set for each embedded module.
+#
+# Standard modules source tree: <scp-path>/module/<name>/src/mod_<name>.c
+# Optee modules source tree:    <scp-path>/module/optee/<short-name>/src/mod_<name>.c
+# Product modules source tree:  <scp-path>/product/<product-name>/module/<name>/src/mod_<name>.c
+#
+# scpfw-embed-generic-module is to be used for generic modules.
+# scpfw-embed-optee-module is to be used for optee modules.
+# scpfw-embed-product-module is to be used for product modules.
+# Modules that implements other C source files aside mandate mod_<name>.c must
+# be explicitly added in srcs-y, as for generic module scmi below.
+#
+# scpfw-embed-mod takes 4 arguments:
+# $1 module name, lowercase
+# $2 module directory name
+# $3 module parent directory relative path in scpfw tree
+# $4 module name, uppercase, relates to _CFG_SCPFW_MOD_$4
+define scpfw-embed-mod
+ifneq (,$$(wildcard $(scpfw-path)/$3/$2/include/*))
+incdirs_ext-y += $(scpfw-path)/$3/$2/include
```

Comment:
This could lead to some very long command lines. This is more of an observation, I'm not sure it's a problem.